src/main/editDelete/liveProcess.ts

```diff
@@ -0,0 +1,42 @@
+import { Transcription, Word } from 'sharedTypes';
+
+/**
+ * calculateTime calculates the outputStartTimes of a word based on the
+ * current state of all words in a Transcript. It takes into account edits/deletes
+ * @param word word in Transcript
+ * @param i the current index position in the words array
+ * @param words an array of words from Transcript
+ * @returns the input word with an updated outputStartTime
+ *
+ */
+
+const calculateTime = (word: Word, i: number, words: Word[]): Word => {
+  if (!word.deleted) {
+    if (i === 0) {
+      word.outputStartTime = 0;
+    } else {
+      let j = i - 1;
+      while (j > -1 && words[j].deleted) {
+        j -= 1;
+      }
+      if (j === -1) {
+        word.outputStartTime = 0;
+      } else {
+        word.outputStartTime = words[j].outputStartTime + words[j].duration;
```

Comment:
Notice that you're only ever updating the outputStartTime, so how could you redesign this function to just return the outputStartTime you want?

Reply:
This is the only change request I'm not 100% sure about. I've refactored calculateTime to return the updated outputStartTime, but I now have to call another function processWord for the mapping. If there's any better suggestions please let me know and I'll try implement it.